vcd/resource_vcd_nsxt_edgegateway_l2_vpn_tunnel.go

```diff
@@ -0,0 +1,399 @@
+)
+
+func resourceVcdNsxtEdgegatewayL2VpnTunnel() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceVcdNsxtEdgegatewayL2VpnTunnelCreate,
+		ReadContext:   resourceVcdNsxtEdgegatewayL2VpnTunnelRead,
+		UpdateContext: resourceVcdNsxtEdgegatewayL2VpnTunnelUpdate,
+		DeleteContext: resourceVcdNsxtEdgegatewayL2VpnTunnelDestroy,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceVcdNsxtEdgegatewayL2VpnTunnelImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"org": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Description: "The name of organization to use, optional if defined at provider " +
+					"level. Useful when connected as sysadmin working across different organizations",
+			},
+			"edge_gateway_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Edge gateway ID for the tunnel",
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Name of the L2 VPN Tunnel session",
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Description of the L2 VPN Tunnel session",
+			},
+			"session_mode": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{"CLIENT", "SERVER"}, false),
+				ForceNew:     true,
+				Description:  "Mode of the tunnel session, must be CLIENT or SERVER",
+			},
+			"enabled": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				Description: "Status of the L2 VPN Tunnel session. Always set to `true` for CLIENT sessions",
+			},
+			"local_endpoint_ip": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "Local endpoint IP of the tunnel session, the IP must be sub-allocated to the Edge Gateway",
+				ValidateFunc: validation.IsIPAddress,
+			},
+			"remote_endpoint_ip": {
+				Type:     schema.TypeString,
+				Required: true,
+				Description: "The IP address of the remote endpoint, which corresponds to the device" +
+					"on the remote site terminating the VPN tunnel.",
+				ValidateFunc: validation.IsIPAddress,
+			},
+			"tunnel_interface": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				Description: "Network CIDR block over which the session interfaces. Only relevant if " +
+					"`session_mode` is set to `SERVER`",
+				ValidateFunc: validation.IsCIDR,
+			},
+			"connector_initiation_mode": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Description: "Connector initation mode of the session describing how a connection is made. " +
+					"Needs to be set only if `session_mode` is set to `SERVER`",
+				ValidateFunc: validation.StringInSlice([]string{"INITIATOR", "RESPOND_ONLY", "ON_DEMAND"}, false),
+			},
+			"pre_shared_key": {
+				Type:     schema.TypeString,
+				Optional: true,
```

Comment:
Same remark as done for this field in the data source.

Reply:
I simplified the description here to reduce confusion as the user probably has an idea of what they're creating, also specified when it's required